system-tests-new/fortran-prebuild/test_fortran_prebuild.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import zlib
+from pathlib import Path
+from unittest import mock
+
+from fab.util import file_walk
+
+from fab.build_config import BuildConfig
+from fab.steps.analyse import Analyse
+from fab.steps.compile_fortran import CompileFortran
+from fab.steps.grab import GrabFolder, GrabPreBuild
+from fab.steps.link_exe import LinkExe
+from fab.steps.preprocess import fortran_preprocessor
+from fab.steps.walk_source import FindSourceFiles
+
+
+class TestFortranPrebuild(object):
+
+    def build_config(self, fab_workspace, prebuild_folder=None):
+        logging.getLogger('fab').setLevel(logging.WARNING)
+
+        build_config = BuildConfig(
+            project_label='test_prebuild',
+            fab_workspace=fab_workspace,
+            steps=[
+                GrabFolder(Path(__file__).parent / 'project-source', dst='src'),
+                *([GrabPreBuild(prebuild_folder)] if prebuild_folder else []),
```

Comment:
A bit clearer this way?

```suggestion
                [GrabPreBuild(f) for f in prebuild_folder],
```

Reply:
Hi, thanks but `prebuild_folder` is a path or `None`.
I'll make sure that's more clear in the docstring and type hints.

This is what we're trying to achieve with this line; it's inserting an element (or not) into a list of steps.
```
>>> [1, *[2], 3]
[1, 2, 3]
>>> [1, *[], 3]
[1, 3]
```